br/pkg/storage/azblob.go

```diff
@@ -303,9 +327,29 @@ func newAzureBlobStorageWithClientBuilder(ctx context.Context, options *backuppb
 		return nil, errors.Trace(err)
 	}
 
-	if len(options.EncryptionScope) > 0 && len(options.StorageClass) > 0 {
-		return nil, errors.Errorf("Set Blob Tier cannot be used with customer-provided keys. " +
-			"Please don't supply the access-tier when use encryption-scope.")
+	if (len(options.EncryptionScope) > 0 || options.EncryptionKey != nil) && len(options.StorageClass) > 0 {
+		return nil, errors.Errorf("Set Blob Tier cannot be used with customer-provided key/scope. " +
+			"Please don't supply the access-tier when use encryption-key or encryption-scope.")
+	} else if len(options.EncryptionScope) > 0 && options.EncryptionKey != nil {
```

Comment:
It's better to handle the this automatically. because we get EncryptionKey from env